internal/services/compute/disk_encryption_set_resource.go

```diff
@@ -253,19 +310,24 @@ func resourceDiskEncryptionSetUpdate(d *pluginsdk.ResourceData, meta interface{}
 		update.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
 	}
 
+	keyVaultKeyId := d.Get("key_vault_key_id").(string)
+
+	if keyVaultKeyId == "" {
+		return fmt.Errorf("missing required argument. The argument %q is required, but no definition was found", "key_vault_key_id")
+	}
```

Comment:
As above, this should be `Required` in schema, and but we can check below for a non-empty string when `auto_key_rotation_enabled` is `true`. The following could be used to facilitate this, and use `keyVaultKeyId.ID()` as needed? 
```suggestion
		keyVaultKeyId := parse.ParseOptionallyVersionedNestedItemID(d.Get("key_vault_key_id").(string))
```

Reply:
Fixed.